code-challenges.test.js

```diff
@@ -16,19 +16,50 @@
 
 const fibonacciLength1 = 6
 // Expected output: [1, 1, 2, 3, 5, 8]
-
-const fibonacciLength2 = 10
+  describe("fibonacciLength1",() => {
+    it ("returns an array containing the Fibonacci sequence. The length of the array is determined by the argument of the function", () => {
+      expect(fibonnaci(fibonacciLength1)).toEqual([1, 1, 2, 3, 5, 8])
+    })
+  })    
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
-
+const fibonacciLength2 = 10
 // b) Create the function that makes the test pass.
+describe("fibonacciLength2",() => {
+  it ("returns an array containing the Fibonacci sequence. The length of the array is determined by the argument of the function", () => {
+    expect(fibonnaci(fibonacciLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+  })
+})
+// i was able to get the test statements easily and the else if statements.
+
+function fibonnaci(length) {
+  if (length <= 0) {
+    // if the length comes back as 0 return an empty array
+    return []; //[] represent an empty array
+  } else if (length === 1) { //if the length comes back as stricktly equal to 1 
+    return [1]; // return an array with the first fib
+  } else if (length === 2) { // if the length comes back as stricktly equal to 2
+  return [1, 1]; // return the first two fib
+}
+// This is where i was completely stuck, i knew that we had to use to variables in order for us to achieve the next number but i just didnt know how to code it.
```

Comment:
It looks like you spent a long time on the edge cases that weren't required in the prompt. Fibonacci is a tough one because it takes some time to understand the logic. The next number is the sum of the previous two numbers. In an array, that would be the previous two values.